src/bloomfilter.h

```diff
@@ -52,32 +52,42 @@ BTYPE _hash_long64(uint32_t hash_seed, Key * key);
 
 static inline int bloomfilter_Add(BloomFilter * bf, Key * key)
 {
-    BTYPE (*_hash_char)(uint32_t, Key *) = _hash_char32;
-    BTYPE (*_hash_long)(uint32_t, Key *) = _hash_long32;
     register BTYPE mod = bf->array->bits;
     register int i;
     register int result = 1;
     register BTYPE hash_res;
 
     if (bf->bf_version >= 2 && mod >= HASH64_THRESHOLD) {
-        _hash_char = _hash_char64;
-        _hash_long = _hash_long64;
-    }
-
-    BTYPE (*hashfunc)(uint32_t, Key *) = _hash_char;
-
-    if (key->shash == NULL)
-        hashfunc = _hash_long;
-
-    for (i = bf->num_hashes - 1; i >= 0; --i) {
-        hash_res = (*hashfunc)(bf->hash_seeds[i], key) % mod;
-        if (result && !mbarray_Test(bf->array, hash_res)) {
-            result = 0;
-        }
-        if (mbarray_Set(bf->array, hash_res)) {
-            return 2;
+        BTYPE (*hashfunc)(uint32_t, Key *) = _hash_char32;
```

Comment:
The whole point of the if and copypasting the loop, was to avoid this indirection.